### PR TITLE
Closes #6756 Increase phpstan level 2

### LINKED ISSUE #6756: Increase PHPStan level to 2
[#6621](https://github.com/wp-media/wp-rocket/issues/6621) introduced PHPStan level 1, 
It would be nice to increase the level to 2. 

Once this is resolved, another issue should be created to bump the rule level again.

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #6756

### Technical documentation

Some tests were marked as skipped because they do not perform assertions in their current form. They need to be revisited.

Notable changes:
- Return default values if the user data could not be fetched (settings dashboard)
- Make sure the `label` property is correctly set before using it when adding LCP data from AJAX request
- Bail-out early if the `src` property of the DB entry is empty when adding the `fetchpriority` attribute to an image

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality).

# Checklists

## Feature validation

- [x] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [x] I validated the repo-specific guidelines from CONTRIBUTING.md.

### CODE REVIEW

File: inc/Dependencies/RocketLazyload/Assets.php [hunk 1]
```diff
@@ -267,20 +267,7 @@ public function getYoutubeThumbnailScript( $args = [] ) {
...
+		return "<script>function lazyLoadThumb(e,alt){var t='{$image}',a='<button class=\"play\" aria-label=\"play Youtube video\"></button>';t=t.replace('alt=\"\"','alt=\"'+alt+'\"');return t.replace(\"ID\",e)+a}function lazyLoadYoutubeIframe(){var e=document.createElement(\"iframe\"),t=\"ID?autoplay=1\";t+=0===this.parentNode.dataset.query.length?'':'&'+this.parentNode.dataset.query;e.setAttribute(\"src\",t.replace(\"ID\",this.parentNode.dataset.src)),e.setAttribute(\"frameborder\",\"0\"),e.setAttribute(\"allowfullscreen\",\"1\"),e.setAttribute(\"allow\", \"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\"),this.parentNode.parentNode.replaceChild(e,this.parentNode)}document.addEventListener(\"DOMContentLoaded\",function(){var e,t,p,a=document.getElementsByClassName(\"rll-youtube-player\");for(t=0;t<a.length;t++)e=document.createElement(\"div\"),e.setAttribute(\"data-id\",a[t].dataset.id),e.setAttribute(\"data-query\", a[t].dataset.query),e.setAttribute(\"data-src\", a[t].dataset.src),e.innerHTML=lazyLoadThumb(a[t].dataset.id,a[t].dataset.alt),a[t].appendChild(e),p=e.querySelector('.play'),p.onclick=lazyLoadYoutubeIframe});</script>";
```
Comment (remyperona) [inline inc/Dependencies/RocketLazyload/Assets.php]: @Miraeld why was this changed?

File: inc/classes/dependencies/mobiledetect/mobiledetectlib/Mobile_Detect.php
Comment (remyperona) [inline inc/classes/dependencies/mobiledetect/mobiledetectlib/Mobile_Detect.php:1]: Don't think this should be part of the update

**Review by remyperona — APPROVED** [review]

**Review by hanna-meda — APPROVED** [review]
Ran smoke & lcp tests on e2e for this branch, all passed.
Executed TP with following report:
[testrail-report-619.pdf](https://github.com/user-attachments/files/16286856/testrail-report-619.pdf)
One TC is failed but looks like failing on trunk too. Will be addressed separately.
